x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_bulk_route.ts

```diff
@@ -48,104 +48,112 @@ export const createCreateRulesBulkRoute = (server: ServerFacade): Hapi.ServerRou
         return headers.response().code(404);
       }
 
-      const mappedDuplicates = countBy('rule_id', request.payload);
+      const ruleDefinitions = request.payload;
+      const mappedDuplicates = countBy('rule_id', ruleDefinitions);
       const dupes = getDuplicates(mappedDuplicates);
 
       const rules = await Promise.all(
-        request.payload.map(async payloadRule => {
-          const {
-            description,
-            enabled,
-            false_positives: falsePositives,
-            from,
-            query,
-            language,
-            output_index: outputIndex,
-            saved_id: savedId,
-            meta,
-            filters,
-            rule_id: ruleId,
-            index,
-            interval,
-            max_signals: maxSignals,
-            risk_score: riskScore,
-            name,
-            severity,
-            tags,
-            threat,
-            to,
-            type,
-            references,
-            timeline_id: timelineId,
-            timeline_title: timelineTitle,
-            version,
-          } = payloadRule;
-          if (ruleId != null && dupes?.includes(ruleId)) {
-            return createBulkErrorObject({
-              ruleId,
-              statusCode: 409,
-              message: `rule_id: "${ruleId}" already exists`,
-            });
-          }
-          const ruleIdOrUuid = ruleId ?? uuid.v4();
-          try {
-            const finalIndex = outputIndex != null ? outputIndex : getIndex(request, server);
-            const callWithRequest = callWithRequestFactory(request, server);
-            const indexExists = await getIndexExists(callWithRequest, finalIndex);
-            if (!indexExists) {
-              return createBulkErrorObject({
-                ruleId: ruleIdOrUuid,
-                statusCode: 400,
-                message: `To create a rule, the index must exist first. Index ${finalIndex} does not exist`,
-              });
-            }
-            if (ruleId != null) {
-              const rule = await readRules({ alertsClient, ruleId });
-              if (rule != null) {
-                return createBulkErrorObject({
-                  ruleId,
-                  statusCode: 409,
-                  message: `rule_id: "${ruleId}" already exists`,
-                });
-              }
-            }
-            const createdRule = await createRules({
-              alertsClient,
-              actionsClient,
+        ruleDefinitions
+          .filter(rule => !dupes?.includes(rule.rule_id ?? ''))
```

Comment:
What's the `?? ''` addressing? Could `dupes` potentially contain `undefined`, but not `''` ?

Reply:
rule_id can be null on a rule definition in the request payload and `includes` yells at me because it's expecting a string.

Reply:
rule_id can be null, so you are using `''` as a sentinel value. You can do this maybe?

```ts
 .filter(rule => rule.rule_id != null && !dupes?.includes(rule.rule_id))
```

And that might be more exact for what you want instead of using an empty string for a sentinel value.

Reply:
I think this would prevent rules from being created that have an empty rule_id, where we currently allow empty / null rule_ids which we just generate one https://github.com/elastic/kibana/blob/ad568922a1136135e4ed43826ac33a28523e8e60/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_bulk_route.ts#L86

Reply:
I misread that logic. I think that will work. Thanks @FrankHassanabad !